### ISSUE TITLE
Alerts keep sending for deleted rule

### ISSUE THREAD

**Issue by surfer190** [ServerCentral/praeco#165]
It is hard to diagnose from this distributed system, but after deleting a rule from the Praeco frontend. On `v0.45` the alerts keep sending.

Perhaps this is to do with `ApScheduler`.

**Comment by johnsusek** [comment]
Could be. In this case, restarting elastalert should clear it.

**Comment by surfer190** [comment]
Yes, did a `docker-compose restart` and that sorted it.
Is there anything we can do to ensure the rule is removed from elastalert from the praeco side?

**Comment by johnsusek** [comment]
Not really sure about this. We simply delete the rule off the disk. And elastalert is supposed to detect the rule was deleted and stop running it. 

We could do something like restart elastalert everytime a rule is deleted, but that feels too hacky.

I'm going to leave this issue open even though it seems like an elastalert behavior, because it's annoying and I want to fix it

**Comment by surfer190** [comment]
> We could do something like restart elastalert everytime a rule is deleted, but that feels too hacky.

Yes, agreed. Yeah we'll need to do a bit more digging.

**Comment by daichi703n** [comment]
Hi, @surfer190 
I have had same issue, and I temporary fixed `elastalert.py`. Would you try my fixed version?
https://github.com/daichi703n/praeco/compare/50c85e0e9655a8dd75ed60193451c53e963a6587...fix/elastalert2.0.1-is_enabled

```
git clone https://github.com/daichi703n/praeco
cd praeco
export PRAECO_ELASTICSEARCH=<ES FQDN or IP>
```

Hi @johnsusek 
Thanks for great work about Praeco!!

I also comment on ElastAlert Issue. 
https://github.com/Yelp/elastalert/issues/1677#issuecomment-585012044
I don't create PR yet, because my fixes are poor and redundant...

This fix on ElastAlert seems to be taken long time... Do you have any plan of workaround?
If you have time, please test my code above and consider to adding like this temporary fix to `servercentral/elastalert` Docker Image.

Regards,
Daichi

**Comment by johnsusek** [comment]
Does anyone know if this is fixed in upstream yet?

**Comment by johnsusek** [comment]
Noticed qmando said this in the upstream issue: 
"One workaround right now is instead of adding is_enabled: false, rename the file whatever.yaml.disabled or similar."

So I might do something like this for enabling/disabling rules, and deleting rules. essentially rename them first, then delete or disable.

**Comment by surfer190** [comment]
John, I like the rename method.

**Comment by iamk-karthik** [comment]
I confirm the issue "Alerts keep sending for deleted rule" with praeco version 1.1.0.

**Comment by nsano-rururu** [comment]
@iamk-karthik

It's not a praeco issue, it's an elastalert issue. Do not mistake the cause.

**Comment by iamk-karthik** [comment]
> @iamk-karthik
> 
> It's not a praeco issue, it's an elastalert issue. Do not mistake the cause.

Yes, i understand I am renaming the rule as **rule.yaml** into **rule.yaml.disabled** as a workaround.   Even in some cases I do restart elastalert to resolve the issue(because renaming sometimes not working as expected and also adding error message like **rule not found** continuously on each successive rule execution).

**Comment by nsano-rururu** [comment]
https://elastalert2.readthedocs.io/en/latest/running_elastalert.html#disabling-a-rule
>Disabling a Rule
To stop a rule from executing, add or adjust the is_enabled option inside the rule’s YAML file to false. When ElastAlert reloads the rules it will detect that the rule has been disabled and prevent it from executing. The rule reload interval defaults to 5 minutes but can be adjusted via the run_every configuration option.
>
>Optionally, once a rule has been disabled it is safe to remove the rule file, if there is no intention of re-activating the rule. However, be aware that removing a rule file without first disabling it will _not_ disable the rule!
